src/Hangfire.Mongo.Tests/MongoDistributedLockFacts.cs

```diff
@@ -112,20 +114,21 @@ public void Ctor_WaitForLock_SignaledAtLockRelease()
         {
             UseConnection(database =>
             {
-                Task.Run(() =>
+                var t = new Thread(() =>
```

Comment:
@MarLoe wait, I’ve never said that :)

`AsyncLocal<>` **is** supported on .net core. It is not supported on a classic .net before .net 4.6, but `CallContext`-based implementation can be used there.